### DISCUSSION TITLE
What is the meaning of term "scale" in the description of decimal operators?

### DISCUSSION THREAD

**Discussion by LeonidVasilyev** [dotnet/csharplang#1142] [General]
Relevant excerpt from the [description of decimal multiplication](https://github.com/dotnet/csharplang/blob/master/spec/expressions.md#arithmetic-operators):
> The scale of the result, before any rounding, is the sum of the scales of the two operands.

And division:

> The scale of the result is the smallest scale that will preserve a result equal to the nearest representantable decimal value to the true mathematical result.

Description of other operators makes similar statements. To me it isn't clear whether this statement is about what IEEE-754 calls "width" of operands used for actual calculation or "exponent" of the result. .NET Framework documentation currently uses both "scale" (see [Decimal constructor](https://msdn.microsoft.com/en-us/library/bb1c1a6x(v=vs.110).aspx)) and "exponent" (see another [Decimal constructor](https://msdn.microsoft.com/en-us/library/t1de0ya1(v=vs.110).aspx)) for what IEEE-754 defines as "exponent". 

Excerpt from IEEE-754 2.1 Definitions section:
>**2.1.19 exponent:** The component of a finite floating-point representation that signifies the integer power to which the radix is raised in determining the value of that floating-point representation. The exponent e is used when the significand is regarded as an integer digit and fraction field, and the exponent q is used when the significand is regarded as an integer; e=q+p−1 where p is the precision of the format in digits.

>**2.1.55 width of an operation:** The format of the destination of an operation specified by this standard; it will be one of the supported formats provided by an implementation in conformance to this standard.

And 10.3 preferredWidth attributes for expression evaluation section:
> In this standard, a computational operation that returns a numeric result first produces an unrounded result as an exact number of infinite precision. That unrounded result is then rounded to a destination format. For certain language-standard-specified generic operations, that destination format is implied by the widths if the operands and by the preferredWidth attribute currently in effect.

**Comment by svick** [comment]
This due to the fact that `decimal` (unlike `double` and `float`) preserves trailing fractional zeroes. For example, with `double`, `1.0d` and `1.00d` are the same value, while with `decimal`, `1.0m` and `1.00m` are different values. And the difference between the two values is scale.

So, when the description of multiplication says that the scales are summed, this means that for example the result of `1.0m * 1.00m` is `1.000m`.
